### PR TITLE
Adding the ability to add an existing gRPC service

### LINKED ISSUE #301: Adding the capability to mix current gRPC services with the Python SDK gRPC extention
## Describe the proposal
If we can add a current service to the Python SDK gRPC extension server, then it will enable us to easily use gRPC proxying and also use other features that the SDK provides such as pub/sub etc.

### PR DESCRIPTION
**Pull request by omid-verneek**
# Description

I have added `add_external_service` to the gRPC extension. I also provided an example to showcase the feature. 
This PR helps our org to easily use our gRPC based services with the Python SDK. I think it can also help others to use gRPC proxying for invoking their methods. At the same time, the user gets the full capabilities that come with the SDK.

## Issue reference

This will close #301 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

File: ext/dapr-ext-grpc/dapr/ext/grpc/app.py
Comment (halspang) [inline ext/dapr-ext-grpc/dapr/ext/grpc/app.py:47]: Is this for hosting the gRPC service?

File: examples/grpc_proxying/invoke-reciever.py
Comment (halspang) [inline examples/grpc_proxying/invoke-reciever.py:26]: Commented code?

File: examples/grpc_proxying/invoke-caller.py
Comment (halspang) [inline examples/grpc_proxying/invoke-caller.py:12]: Will this be necessary for every call that the user makes to their gRPC service? Is it possible to provide this automatically? Perhaps when you construct the client?

**Review by halspang — COMMENTED** [review]
Thanks for your submission! 

My knowledge of python is a little limited with gRPC, but the main comment I have is if we can provide the `dapr-app-id` header automatically when making calls. Requiring the user to provide it makes it feel a little clunky.

Here's what the .NET SDK did for reference: https://github.com/dapr/dotnet-sdk/pull/755

File: ext/dapr-ext-grpc/dapr/ext/grpc/app.py
Comment (omid-verneek) [reply ext/dapr-ext-grpc/dapr/ext/grpc/app.py:47]: yes

File: examples/grpc_proxying/invoke-caller.py
Comment (omid-verneek) [reply examples/grpc_proxying/invoke-caller.py:11]: So the idea here is to follow gRPC proxying: [You can see here](https://docs.dapr.io/developing-applications/building-blocks/service-invocation/howto-invoke-services-grpc/)
This way a client can just use their grpc client to call the Dapr sidecar without using the client directly and Dapr will redirect it to the right service. All only by adding metadata.

File: examples/grpc_proxying/README.md
Comment (wcs1only) [inline examples/grpc_proxying/README.md:24]: So these step annotation comments are not just for show, we actually have some automation which validates our examples with every run. It is capable of running your \`\`\`bash ... \`\`\` codeblocks and checking stdout. This avoids the somewhat tedious process of manually running the examples to make sure they work with every release.

Here's an example of what this looks like:
https://github.com/dapr/python-sdk/blame/master/examples/invoke-binding/README.md#L41-L47

Would you be willing to add these steps for at least the standalone (non-k8s) portion of your examples? You don't have to validate all of the output of your example, just enough to say "this example is working".

Here's the detailed guide for our example automation tool:
https://github.com/dapr/mechanical-markdown/tree/main/examples

Also, if you add your new example here, it will get run on every PR:
https://github.com/dapr/python-sdk/blob/master/tox.ini#L36-L45

To run validation locally:
```
pip3 install mechanical-markdown

cd examples/grpc_proxying
mm.py -l README.md
```

**Review by wcs1only — COMMENTED** [review]
You have one minor lint error:
https://github.com/dapr/python-sdk/runs/4241882285?check_suite_focus=true#step:5:118

If you'd like to run the linter locally without waiting for our CI/CD, you can just run:
```
tox -e flake8
```

File: examples/grpc_proxying/README.md
Comment (omid-verneek) [reply examples/grpc_proxying/README.md:24]: let me give it a shot

Comment (omid-verneek) [reply examples/grpc_proxying/README.md]: @wcs1only should be good now

File: examples/grpc_proxying/helloworld_service_pb2_grpc.py
Comment (berndverst) [inline examples/grpc_proxying/helloworld_service_pb2_grpc.py:66]: I'm a bit concerned about this breaking our automated tests -- specifically the `validate examples` workflow we run via Mechnical Markdown.

We will certainly be updating the GRPC library as new versions come out. I assume given the experimental nature of this feature the maintainers of the GRPC library may be moving / removing this feature in the future.

Is there a different way to accomplish what you need to accomplish without the use of an experimental API?

Comment (omid-verneek) [reply examples/grpc_proxying/helloworld_service_pb2_grpc.py:66]: This is what is currently being generated using gRPC tools. I don't have control over this part. I have to mention this is part of the example and not part of the gRPC extension.

Comment (wcs1only) [reply examples/grpc_proxying/helloworld_service_pb2_grpc.py:66]: Yeah, this is something internal to the generated code. The main dapr API's grpc generated code has this too.

**Review by wcs1only — APPROVED** [review]
Awesome, thanks for your contribution.

**Pull request merged**
